src/training_controller/training_controller.py

```diff
@@ -139,16 +138,15 @@ async def es_training_signal_coroutine(signals_queue: asyncio.Queue):
 def job_not_currently_running(job_name, namespace=NAMESPACE):
     try:
         jobs = api_instance.list_namespaced_job(namespace, timeout_seconds=60)
+        for job in jobs.items:
+            if job.metadata.name == job_name:
+                return False
+        return True
```

Comment:
should this function return False if it drops into exception condition?

Reply:
it doesn't need to return anything if it falls into an exception.
I refactored because this block of code was being executed outside of the try-except block, which could lead to a scenario in which `jobs` will not exist